lab/src/raft/leader_election.go

```diff
@@ -19,6 +25,40 @@ func (rf *Raft) runLeaderElection() {
 	// TODO(ling): Implement vote
 }
 
+func (rf *Raft) sendHeartbeat() {
+	lastLog := getLastLog(rf.log)
+	args := AppendEntriesArgs{
+		Term:             rf.currentTerm,
+		Leader:           rf.me,
+		PrevLogIndex:     lastLog.Index,
+		PrevLogTerm:      lastLog.Term,
+		Entries:          []LogEntry{},
+		LeaderCommit:     rf.commitIndex,
+	}
+	reply := AppendEntriesReply{}
+	for i := range rf.peers {
+		if i != rf.me {
+			rf.sendAppendEntries(i, &args, &reply)
+			if reply.Success {
+				// If successful: update nextIndex and matchIndex for follower (§5.3)
```

Comment:
This should be done inside appendEntries

Reply:
Done.